youtube_dl/extractor/archiveorg.py

```diff
@@ -40,9 +40,12 @@ def _real_extract(self, url):
         video_id = self._match_id(url)
         webpage = self._download_webpage(
             'http://archive.org/embed/' + video_id, video_id)
+        input_element_with_playlist = self._search_regex(
+            r'(<\s*input.*\s*class\s*=\s*[\'"].*\s*js-play8-playlist\s*.*[\'"]\s*.*>)',
+            webpage, 'jwplayer playlist')
         jwplayer_playlist = self._parse_json(self._search_regex(
-            r"(?s)Play\('[^']+'\s*,\s*(\[.+\])\s*,\s*{.*?}\)",
-            webpage, 'jwplayer playlist'), video_id)
+            r'.*\s+value\s*=\s*[\'"](.+)[\'"][\s/]',
+            input_element_with_playlist, 'playlist data'), video_id)
```

Comment:
`extract_attributes`.

Reply:
Thanks for the pointer. I changed the code to use extract_attributes() in commit e910f49.

It was still bugging me that I couldn't use get_element_by_class() in the line before as it only returns the *content* of the element and not the element (as the name of the function indicates). So I changed get_element_by_class() (and it's associated functions) in commit b98d1c0 to accept an optional `include_tag` parameter. By default it's set to `False`. So no existing code will be affected by it.

This should prove useful in the future and make those functions more intuitive to use.

I also added tests to `test_utils.py`. While doing so I noticed that get_element_by_class() didn't work for class names starting with a hyphen. I fixed that, too.